### DISCUSSION TITLE
New version of xTool Creative Space

### DISCUSSION THREAD

**Discussion by ulab** [Homebrew/discussions#5703] [Casks] [answered]
### Output of `brew config`

```shell
irrelevant
```

### Output of `brew doctor`

```shell
irrelevant
```

### Description of issue

Hiho,

xTool Creative Space is available in version 2.2.23, but I am not sure how do modify the cask file, especially the livecheck.

URL to check is now the [:homepage](https://www.xtool.com/pages/software), but one issue is that the intel version does not have an arch tag in the filename anymore. 

```
https://storage-us.xtool.com/resource/efficacy/xcs/prod-us/packages/28/2e1bddec-0d1a-4dab-bfd5-de1b68cacd35/xTool-Creative-Space-2.2.23-2024-10-11-11-50-02-arm64.dmg
https://storage-us.xtool.com/resource/efficacy/xcs/prod-us/packages/16/0fbeb19a-b594-4f86-a200-e5e736e6f038/xTool-Creative-Space-2.2.23-2024-10-11-11-24-25.dmg
```

I am not sure how to handle that in the cask file `url` and the livecheck. Perhaps someone else can look into it?

```
  on_arm do
    version "2.2.23,28,2e1bddec-0d1a-4dab-bfd5-de1b68cacd35,2024-10-11-11-50-02"
    sha256  "d928d3e71e25ded28b208547167dd1ab8753f1a1e8935fc19091d6a8fec38097"
  end
  on_intel do
    version "2.2.23,16,0fbeb19a-b594-4f86-a200-e5e736e6f038,2024-10-11-11-24-25"
    sha256 "4bdd6b51d35974590b75bc9409bbb422fad85f76474923669a8d5524058198bb"
  end
  ```

Livecheck probably needs to be switched to `pageMatch`. The regex should still work fine, besides the missing arch tag.

### Relevant casks

xtool-creative-space.rb

**Comment by daeho-ro** [comment, marked as answer]
Don't worry, just open a PR. Then others will comment about that!

**Reply by stianmarthinsen0702-ctrl** [reply]
https://github.com/orgs/Homebrew/discussions/5703#discussioncomment-11118128

**Comment by stianmarthinsen0702-ctrl** [comment]
on_arm do
    version "2.2.23,28,2e1bddec-0d1a-4dab-bfd5-de1b68cacd35,2024-10-11-11-50-02"
    sha256  "d928d3e71e25ded28b208547167dd1ab8753f1a1e8935fc19091d6a8fec38097"
  end
  on_intel do
    version "2.2.23,16,0fbeb19a-b594-4f86-a200-e5e736e6f038,2024-10-11-11-24-25"
    sha256 "4bdd6b51d35974590b75bc9409bbb422fad85f76474923669a8d5524058198bb"
  end
